src/AffineMap.jl

```diff
@@ -0,0 +1,365 @@
+
+julia> ∈([5.0, 1.0], am)
+false
+
+julia> ∈([3.0, 1.0], am)
+true
+```
+
+An example with a non-square matrix:
+
+```jldoctest
+julia> B = BallInf(zeros(4), 1.);
+
+julia> M = [1. 0 0 0; 0 1 0 0]/2;
+
+julia> ∈([0.5, 0.5], M*B)
+true
+```
+"""
+function ∈(x::AbstractVector{N}, am::AffineMap{N})::Bool where {N<:Real}
+    return ∈(am.M \ (x - am.b), am.X)
+end
+
+"""
+    vertices_list(am::AffineMap{N};
+                  [apply_convex_hull]::Bool)::Vector{Vector{N}} where {N<:Real}
+
+Return the list of vertices of a (polyhedral) affine map.
+
+### Input
+
+- `am`                -- affine map
+- `apply_convex_hull` -- (optional, default: `true`) if `true`, apply the convex
+                         hull operation to the list of vertices transformed by the
+                         affine map 
+
+### Output
+
+A list of vertices.
+
+### Algorithm
+
+This implementation computes all vertices of `X`, then transforms them through
+the affine map, i.e.  `v ↦ M*v + b` for each vertex `v` of `X`. By default, the
+convex hull operation is taken before returning this list. For dimensions three
+or higher, this operation relies on the functionality through the concrete
+polyhedra library `Polyhedra.jl`.
+
+If you are not interested in taking the convex hull of the resulting vertices under
+the affine map, pass `apply_convex_hull=false` as a keyword argument.
+
+Note that we assume that the underlying set `X` is polyhedral, either concretely
+or lazily, i.e. there the function `vertices_list` should be applicable.
+"""
+function vertices_list(am::AffineMap{N};
+                       apply_convex_hull::Bool=true)::Vector{Vector{N}} where {N<:Real}
+    # for a zero map, the result is just the singleton containing the translation
+    if iszero(am.M)
+        return Singleton(am.b)
+    end
+
+    # collect low-dimensional vertices lists
+    vlist_X = vertices_list(am.X)
+
+    # create resulting vertices list
+    vlist = Vector{Vector{N}}()
+    sizehint!(vlist, length(vlist_X))
+    for v in vlist_X
+        push!(vlist, am.M * v + am.b)
+    end
+
+    return apply_convex_hull ? convex_hull(vlist) : vlist
+end
+
+"""
+    constraints_list(am::AffineMap{N}) where {N<:Real}
+
+Return the list of constraints of a (polyhedral) affine map.
+
+### Input
+
+- `am` -- affine map
+
+### Output
+
+The list of constraints of the affine map.
+
+### Notes
+
+We assume that the underlying set `X` is polyhedral, i.e., offers a method
+`constraints_list(X)`.
+
+### Algorithm
+
+Falls back to the list of constraints of the translation of a lazy linear map.
+"""
+function constraints_list(am::AffineMap{N}) where {N<:Real}
+    return constraints_list(LinearMap(am.M, am.X) ⊕ am.b)
+end
+
+"""
+    linear_map(M::AbstractMatrix{N}, am::AffineMap{N}) where {N<:Real}
+
+Return the linear map of a lazy affine map.
+
+### Input
+
+- `M`  -- matrix
+- `am` -- affine map
+
+### Output
+
+The polytope representing the linear map of the lazy affine map of a set.  
```

Comment:
Does it have to be a polytope?

Reply:
Oh, good catch, my implementation is wrong here -- it is now returning a lazy translation, but the aim was that `linear_map` returns a concrete set..